genologics/entities.py

```diff
@@ -949,12 +993,30 @@ def get_next_actions(self):
         return actions
 
     def set_next_actions(self, actions):
+        self.get()
         for node in self.root.find('next-actions').findall('next-action'):
             art_uri = node.attrib.get('artifact-uri')
-            action = [action for action in actions if action['artifact'].uri == art_uri][0]
-            if 'action' in action: node.attrib['action'] = action.get('action')
+            action_opts = [action for action in actions if action['artifact'].uri == art_uri][0]
+            if 'action' not in action_opts:
+                raise RuntimeError("No action type specified: {}".format(action_opts))
+            elif action_opts['action'] not in self._action_attrs:
+                raise RuntimeError("Invalid action type specified: {}".format(action_opts))
+
+            node.attrib['action'] = action_opts['action']
+            if action_opts['action'] == 'nextstep':
```

Comment:
Should this be removed so all types of actions can be set? However, 'review' may need also to set escalation?

Reply:
ah, yeah, that can be pulled out. good catch, I'll update it